### PR TITLE
Move tslint.json advice from PULL_REQUEST_TEMPLATE.md

### PR DESCRIPTION
**Pull request by jablko**
Move the advice from the PULL_REQUEST_TEMPLATE.md to the README, [by way of the bot](https://github.com/DefinitelyTyped/dt-mergebot/pull/223#pullrequestreview-529504577), to explain the required form and what to do about it.
- I created a "tslint.json" section to locate it.
  - I didn't put it under "Edit an existing package" or "Create a new package" because I'd like it (reference documentation for the required form) to stand out from the process of making a pull request.
  - I didn't put it under "Linter" for the same reason: not to intersperse the required form (what it is and how to react to satisfy it) with the development process (how to run the linter, writing tests). Where `tslint.json` is talked about I put a link to the section.
- I used the language from the PULL_REQUEST_TEMPLATE.md because it's more explicit and prescriptive than the README is currently. It states directly what to do, and links to the TSLint docs for background.
- I pulled the items out of the PULL_REQUEST_TEMPLATE.md checklist because the bot now displays a checklist, and will link to the same documentation, selectively (when it applies to the change at hand).
- I added the English text to each of the translations so translators can identify what changed.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (elibarzilay) [inline .github/PULL_REQUEST_TEMPLATE.md:17]: I think that it's good to still keep a reminder in the template.  Instead of removing it, it could be trimmed to be a quick reference to the README file section.

**Review by elibarzilay — CHANGES_REQUESTED** [review]

**Review by sandersn — APPROVED** [review]
I agree with @elibarzilay's suggestions. I think this'll be ready to merge once those are addressed.

File: README.md
Comment (jablko) [reply README.md]: Done :heavy_check_mark:

File: README.md
Comment (jablko) [reply README.md:251]: What do you think about instead combining the "Linter" section with "Verifying"? Although technically `// $ExpectError` and `// $ExpectType` are handled by dtslint vs. tsc, I think conceptually they're associated with writing tests, and not with `tslint.json`?

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (jablko) [reply .github/PULL_REQUEST_TEMPLATE.md:17]: Done :heavy_check_mark:

File: README.md
Comment (elibarzilay) [inline README.md:254]: Needs rephrasing of the first two sentences (the first talks about additions to what's specified in the second).

Also, "If for some reason" is better on a new paragraph.

```suggestion
The liner configuration file, `tslint.json` should contain `{ "extends": "dtslint/dt.json" }`, and no additional rules.

If for some reason some rule needs to be disabled, [disable it for that specific line](https://palantir.github.io/tslint/usage/rule-flags/#comment-flags-in-source-code:~:text=%2F%2F%20tslint%3Adisable%2Dnext%2Dline%3Arule1%20rule2%20rule3...%20%2D%20Disables%20the%20listed%20rules%20for%20the%20next%20line) using `// tslint:disable-next-line:[ruleName]` — not for the whole package, so that disabling can be reviewed.  (There are some legacy lint configs that have additional contents, but these should not happen in new work.)
```

File: README.md
Comment (elibarzilay) [inline README.md:256]: Since linting is closer to the verification, I think that it would be best to move the "Common mistakes" so it's after verification.  (So the order is linting -- verification -- common mistakes.)

Should be updated in the TOC too.

File: README.md
Comment (elibarzilay) [inline README.md:309]:
```suggestion
Use `$ExpectType` to assert that an expression is of a given type, and `$ExpectError` to assert that a compile error.  Examples:
```

Also, maybe mention the "Minimum TypeScript Version" line?

File: README.md
Comment (elibarzilay) [inline README.md:307]: I know that this wasn't here before, but I think that it's worth starting with a quick paragraph saying that tests are preferably in files that look like that (`test` in the name or in the path), and that these files should have the following.

File: README.md
Comment (elibarzilay) [inline README.md:323]: Suggestion (the GH suggestion doesn't work here):

```suggestion
Test your changes by running `npm test package-name` where `package-name` is the name of your package.

This script uses [dtslint](https://github.com/microsoft/dtslint) to run the TypeScript compiler against your files.  See [the description there](https://github.com/Microsoft/dtslint#write-tests) for more details.
```

**Review by elibarzilay — CHANGES_REQUESTED** [review]

File: README.md
Comment (jablko) [reply README.md]: Done :heavy_check_mark:

File: README.md
Comment (jablko) [reply README.md]: Done :heavy_check_mark:

File: README.md
Comment (jablko) [reply README.md]: Done :heavy_check_mark:

File: README.md
Comment (jablko) [reply README.md:266]: Nice, I opened [another PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/49636) for these changes :+1:

**Comment by sheetalkamat** [comment]
@elibarzilay do you want to take a look again before i merge this.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (amcasey) [inline .github/PULL_REQUEST_TEMPLATE.md:17]: I worry that some (many?) users won't click through so I'd like to include the most important bit here: "no disabling rules".  If someone disagrees, they can click through and see more nuanced guidance.

Comment (amcasey) [reply .github/PULL_REQUEST_TEMPLATE.md:17]: Also, "be present" seems redundant.

Comment (jablko) [reply .github/PULL_REQUEST_TEMPLATE.md]: That's smart, done :heavy_check_mark:

**Review by amcasey — APPROVED** [review]
LGTM.  For the record, I did not review the translated docs - I assumed they just had the same reordering as the English docs.

**Review by elibarzilay — APPROVED** [review]

**Review by amcasey — APPROVED** [review]
Looks like the last push reset the bot.

**Comment by rbuckton** [comment]
The bot didn't comment about being able to self merge. Will that still work or do we have to merge this manually?

**Review by rbuckton — APPROVED** [review]

**Pull request merged**
